odoo/orm/environments.py

```diff
@@ -374,13 +387,15 @@ def protecting(self, what, records=None):
         protected = self._protected
         try:
             protected.pushmap()
+            ids_by_field: Mapping[Field, Collection[IdType]]
             if records is not None:  # Handle first signature
-                ids_by_field = {field: records._ids for field in what}
+                ids_by_field = dict.fromkeys(what, records._ids)
             else:  # Handle second signature
-                ids_by_field = defaultdict(list)
+                list_ids_by_field: Mapping[Field, list[IdType]] = defaultdict(list)
                 for fields, what_records in what:
                     for field in fields:
-                        ids_by_field[field].extend(what_records._ids)
+                        list_ids_by_field[field].extend(what_records._ids)
+                ids_by_field = list_ids_by_field
```

Comment:
The typing info here has no added value, since the code is less than 20 lines.  It only makes the code more obscure IMHO.  Please remove it.

I would instead rewrite the code to handle the first signature as the second one:
```py
            if records is not None:  # convert first signature to second one
                what = [(what, records)]
            ids_by_field = defaultdict(list)
            for fields, what_records in what:
                for field in fields:
                    ids_by_field[field].extend(what_records._ids)
```